privacyidea/lib/tokenclass.py

```diff
@@ -449,6 +451,59 @@ def authenticate(self, passw, user=None, options=None):
 
         return pin_match, otp_counter, reply
 
+    @staticmethod
+    def decode_base32check(encoded_otpkey):
```

Comment:
These two static methods: are they so generic, that we could move this to ``lib/utils.py``?

Woulds these ever be used outside of token classes?

Reply:
In fact it is not the otpkey, that is encoded, but the client part.
So we might want to change the naming, so the we do not mislead developers.

This decoding function simply is a out own crc function.
So in fact we could genericly do:

    def decode_base32check(encoded_data):

and talk of encoded_data, not of an otpkey.

...and than we could move it to the utils module... and have extra tests for it! ;-)

Reply:
Hm yeah, I guess moving them to ``lib/utils.py`` is a good idea, as they are not specific to the HOTP token.

I agree that `encoded_otpkey` should be renamed to `encoded_data`.

As the client part is passed in the ``otpkey`` parameter, it actually is an otp key from privacyIDEA's point of view :) A side-effect of this PR is that we can also pass ordinary (non-2step) OTP keys in the base32check format.

Reply:
Great! We are all right! ;-) It is an otpkey, but only in this case. If we move a decode_base32check function to the generic lib/utils, we can use this for any data in the future. Maybe for transferring encryption keys for import files or whatever...

Reply:
ee98ffc377f895843f6d1d777d1f72b72610989e moves the function to ``lib/utils.py``.